lib/client-base.js

```diff
@@ -563,6 +555,14 @@ function resume (client) {
       continue
     }
 
+    if (!client.connected) {
+      return
+    }
```

Comment:
We won't connect if there is no pending request, therefore it would never splice away cancelled requests, therefore it wouldn't close the client.